src/codegen/yul/builtin.rs

```diff
@@ -122,23 +125,43 @@ pub(crate) fn process_builtin(
         | YulBuiltInFunction::CallCode
         | YulBuiltInFunction::DelegateCall
         | YulBuiltInFunction::StaticCall
-        // Return and revert also load from memory, so we first need to solve mload and mstore builtins
-        | YulBuiltInFunction::Return
-        | YulBuiltInFunction::Stop // Stop is the same as return(0, 0)
-        | YulBuiltInFunction::Revert
         // Log functions
         | YulBuiltInFunction::Log0
         | YulBuiltInFunction::Log1
         | YulBuiltInFunction::Log2
         | YulBuiltInFunction::Log3
         | YulBuiltInFunction::Log4
         // origin is the same as tx.origin and is not implemented
-        | YulBuiltInFunction::Origin
-        => {
+        | YulBuiltInFunction::Origin => {
             let function_ty = builtin_ty.get_prototype_info();
             unreachable!("{} yul builtin not implemented", function_ty.name);
         }
 
+        // Return and revert also load from memory, so we first need to solve mload and mstore builtins
+        YulBuiltInFunction::Return => {
+            if ns.target != Target::EVM {
+                let function_ty = builtin_ty.get_prototype_info();
+                unreachable!("{} yul builtin not implemented", function_ty.name);
+            }
+
+            let addr = expression(&args[0], contract_no, ns, vartab, cfg, opt).cast(&Type::Ref(Type::Uint(256).into()), ns);
+            let length = expression(&args[0], contract_no, ns, vartab, cfg, opt).cast(&Type::Uint(256), ns);
+
+            cfg.add(vartab, Instr::Return { value: vec![addr, length] });
+            Expression::Poison
+        }
+        YulBuiltInFunction::Stop // Stop is the same as return(0, 0)
+        | YulBuiltInFunction::Revert => {
```

Comment:
Revert should revert the changes in this call. If we only return, we are not reverting anything.

Reply:
These new implementations must be tested as well.

Reply:
We don't have EVM code gen so these cannot be tested. These were simply added to make the `fn ethereum_solidity_tests()` work.